src/nvim/buffer_defs.h

```diff
@@ -361,14 +361,36 @@ struct mapblock {
   sctx_T m_script_ctx;          // SCTX where map was defined
 };
 
-/*
- * Used for highlighting in the status line.
- */
+/// Used for highlighting in the status line.
+typedef struct stl_hlrec stl_hlrec_t;
 struct stl_hlrec {
   char_u      *start;
   int userhl;                   // 0: no HL, 1-9: User HL, < 0 for syn ID
 };
 
+/// Used for building the status line.
+typedef struct stl_item stl_item_t;
+struct stl_item {
+  // Where the item starts in the status line output buffer
+  char_u *start;
```

Comment:
Patch 8.2.1909 has different struct members. Intentional?

Reply:
Yes, I felt that prefixing everything with `stl_` added a lot of unnecessary noise. That's how I would have kept them if I had contributed the patch in neovim without needing to go upstream. I know there is a precedent of prefixing in eg `win_T` or `buf_T` but here it doesn't feel right. Let me know if they need the prefix.

Reply:
It's fine. Copy/paste your rationale for your changes in the commit so that there is an explanation when the developer runs `git blame`.

Reply:
Done, I've added the mention on the initial commit here.